### PR TITLE
Deserialize custom fields

### PR DESCRIPTION
**Pull request by MicheleRezk**
Using FieldMapperTypeRetriever to deserialize custom fields

### CODE REVIEW

File: commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        
+        [Fact]
+        public void CustomFieldsListDeserialization()
+        {
+            var serializerService = this._serializationFixture.SerializerService;
+            var serialized = File.ReadAllText("Resources/CustomFields/List.json");
+            var deserialized = serializerService.Deserialize<Category>(serialized);
+            Assert.IsType<string>(deserialized.Custom.Fields["string"]);
+            Assert.IsType<double>(deserialized.Custom.Fields["integer"]);
+            Assert.Equal(10.0, deserialized.Custom.Fields["integer"]);
```
Comment (jenschude) [inline commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs:30]:
```suggestion
            Assert.IsType<integer>(deserialized.Custom.Fields["integer"]);
            Assert.Equal(10, deserialized.Custom.Fields["integer"]);
```

File: commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            Assert.IsType<string>(deserialized.Custom.Fields["enum"]);
+            Assert.IsType<string>(deserialized.Custom.Fields["localized-enum"]);
+            Assert.IsType<CategoryReference>(deserialized.Custom.Fields["reference"]);
+            Assert.IsType<CentPrecisionMoney>(deserialized.Custom.Fields["money"]);
+            Assert.IsType<LocalizedString>(deserialized.Custom.Fields["localized-string"]);
+            
+            Assert.IsType<List<Object>>(deserialized.Custom.Fields["set-empty"]);
+            Assert.IsType<List<bool>>(deserialized.Custom.Fields["set-boolean"]);
+            Assert.IsType<List<double>>(deserialized.Custom.Fields["set-integer"]);
+            Assert.Equal(10.0, (deserialized.Custom.Fields["set-integer"] as List<Double>).First());
```
Comment (jenschude) [inline commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs:46]:
```suggestion
            Assert.IsType<List<integer>>(deserialized.Custom.Fields["set-integer"]);
            Assert.Equal(10, (deserialized.Custom.Fields["set-integer"] as List<integer>).First());
```

File: commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        [Fact]
+        public void CustomFieldsSerialization()
+        {
+            var serializerService = this._serializationFixture.SerializerService;
+            var fieldContainer = new FieldContainer
+            {
+                {"int", 13}, {"double", 13.0}, {"double2", 13.2}
+            };
+            var result = serializerService.Serialize(fieldContainer);
+            var expectedResult = "{\"int\":13,\"double\":13,\"double2\":13.2}";
```
Comment (jenschude) [inline commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs:68]:
```suggestion
            var expectedResult = "{\"int\":13,\"double\":13.0,\"double2\":13.2}";
```

File: commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs (hunk 1 above)
Comment (MicheleRezk) [reply commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs]: Done

File: commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs (hunk 2 above)
Comment (MicheleRezk) [reply commercetools.Sdk/Tests/commercetools.Api.Serialization.Tests/CustomFieldsTests.cs]: Done

**Review by jenschude — APPROVED** [review]

**Review by jenschude — APPROVED** [review]

**Pull request merged**